js/src/builtin/streams/ReadableStreamBYOBRequest.cpp

```diff
@@ -0,0 +1,235 @@
+#include "builtin/streams/ClassSpecMacro.h"             // JS_STREAMS_CLASS_SPEC
+#include "builtin/streams/MiscellaneousOperations.h"  // js::TransferArrayBuffer
+#include "builtin/streams/PullIntoDescriptor.h"       // js::PullIntoDescriptor
+#include "builtin/streams/ReadableByteStreamControllerOperations.h"  // js::ReadableByteStreamControllerRespondInternal
+#include "builtin/streams/ReadableStreamController.h"  // js::ReadableByteStreamController
+#include "js/ErrorReport.h"             // JS_ReportErrorNumberASCII
+#include "js/experimental/TypedData.h"  // JS_GetArrayBufferViewData, JS_IsUint8Array
+#include "js/friend/ErrorMessages.h"  // js::GetErrorMessage, JSMSG_*
+#include "js/PropertySpec.h"
+#include "vm/ArrayBufferViewObject.h"  // js::UnwrapArrayBufferView
+#include "vm/PromiseObject.h"          // js::UnwrapAndTypeCheckThis
+
+#include "builtin/streams/ReadableStreamReader-inl.h"
+#include "vm/Compartment-inl.h"  // js::UnwrapAndTypeCheckThis
+#include "vm/Compartment-inl.h"  // JS::Compartment::wrap, js::UnwrapAnd{DowncastObject,TypeCheckThis}
+#include "vm/JSObject-inl.h"      // js::NewObjectWithClassProto
+#include "vm/NativeObject-inl.h"  // js::ThrowIfNotConstructing
+
+using JS::CallArgs;
+using JS::Handle;
+using JS::Rooted;
+using JS::RootedObject;
+using JS::RootedValue;
+using JS::Value;
+
+using js::GetErrorMessage;
+using js::ListObject;
+using js::NewObjectWithClassProto;
+using js::PromiseObject;
+using js::PullIntoDescriptor;
+using js::ReadableByteStreamController;
+using js::ReadableStream;
+using js::ReadableStreamBYOBRequest;
+using js::ReadableStreamReader;
+using js::TransferArrayBuffer;
+using js::UnwrapAndDowncastObject;
+using js::UnwrapAndTypeCheckThis;
+
+bool ReadableStreamBYOBRequest::constructor(JSContext* cx, unsigned argc,
+                                            Value* vp) {
+  JS_ReportErrorNumberASCII(cx, GetErrorMessage, nullptr,
+                            JSMSG_BOGUS_CONSTRUCTOR,
+                            "ReadableStreamBYOBRequest");
+  return true;
+}
+
+// https://streams.spec.whatwg.org/#rs-byob-request-view
+static bool ReadableStreamBYOBRequest_view(JSContext* cx, unsigned argc,
+                                           Value* vp) {
+  CallArgs args = CallArgsFromVp(argc, vp);
+
+  Rooted<ReadableStreamBYOBRequest*> unwrappedRequest(
+      cx,
+      UnwrapAndTypeCheckThis<ReadableStreamBYOBRequest>(cx, args, "get view"));
+  if (!unwrappedRequest) {
+    return false;
+  }
+  args.rval().set(unwrappedRequest->view());
+  return true;
+}
+
+// https://streams.spec.whatwg.org/#rs-byob-request-respond
+[[nodiscard]] static bool ReadableStreamBYOBRequest_respond(JSContext* cx,
+                                                            unsigned argc,
+                                                            Value* vp) {
+  CallArgs args = CallArgsFromVp(argc, vp);
+
+  double bytesWritten = args.get(0).toNumber();
```

Comment:
As this is a required argument, we should first check to see if it was supplied and create a pending exception if it was not supplied


```suggestion
  if (!args.requireAtLeast(cx, "ReadableStreamBYOBRequest.respond", 1)) {
    return false;
  }
  double bytesWritten = args.get(0).toNumber();
```

Reply:
Good catch, thanks.